### ISSUE TITLE
Brotli compression ratio depends on endianness of CPU

### ISSUE THREAD

**Issue by eustas** [google/brotli#580]
TL;DR: Just realized that "unaligned load + mask" technique used by some hashers will give endianness-inconsistent results.

This problem does not affect correctness, only what backward references selection and, consequently, compression ratio.

Lets say, hasher depends on `5` bytes of input. Then, it does the following: `hash = magic * (load64(src) & 0xFFFFFFFFFF)`. On little-endian CPU the first `5` bytes of `src` will affect the resulting hash. On big-endian CPU, the first `3` bytes will be ignored, and next `5` bytes will be taken into account. This way, on big-endian CPU, matches of length `[5..7]` are likely to be missed by hasher. On the other side there will be more `8+` matches.
